test/com/google/javascript/jscomp/PeepholeRemoveDeadCodeTest.java

```diff
@@ -478,6 +481,226 @@ public void testOptimizeSwitch2() {
         "outer: {f(); break outer;}");
   }
 
+  public void testOptimizeSwitchWithLabellessBreak() {
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': var x = 1; break;",
+        "    case 'y': break;",
+        "  }",
+        "}"),
+        "function f() { var x = 1; }");
+
+    test(LINE_JOINER.join(
+        "function f() {",
+        "  switch('x') {",
+        "    case 'x': var x = 1; break; var y = 2;",
+        "    case 'y': break;",
+        "  }",
+        "}"),
+        "function f() { var x = 1; }");
+
+    testSame(LINE_JOINER.join(
+        "function f() {",
+        "  switch(x) {",
```

Comment:
Note, that this is optimizable to
```javascript
if (x == 'y') {
  // case block
} else {
  // default block
}
```
which is generally more optimizable by other passes.